content/blog/2021/08/using-maxsat-for-resolving-deps.md

```diff
@@ -0,0 +1,165 @@
+# Using a MaxSAT for resolving Hypper chart dependencies on Kubernetes
+
+[Hypper](http://hypper.io) is a Kubernetes package manager based on Helm. It
+handles charts in a similar way as Helm does, with a twist: it tries to resolve
+dependencies between charts to reuse existing resources and deployments.
+For more info, see the two articles on this blog by @mattfarina: [Introducing
+Hypper](https://community.suse.com/posts/15893197) and
+[Hypper v3 - Now with a SAT solver](https://community.suse.com/posts/14072167).
+
+Hypper is designed to work with shared dependencies: dependencies installed once
+on a cluster that several charts depend on. Think of Hypper as bringing the
+usual Linux package management to Kubernetes Helm charts.
+
+Checking all the charts so see if their dependencies are correctly installed or
+there's some missing is not an easy task. This is called a "Satisfiability
+problem", or SAT, for friends. In fact, solving a SAT problem it's an [NP-Complete
```

Comment:
ahá! :) In theory it should be "a NP problem", because it's a consonant instead of a vowel, but seems that consonants with vowel sounds include f, h, l, m, n, r, s, and x, particularly in acronyms and so. Examples:
- An NBC television show.
- An HTML document.
- A CBS news outlet.
- A Microsoft program, but An MS-DOS program.

Will leave it as it is :).

Reply:
This did not read smoothly to me. It would read more smoothly as "solving a SAT problem is an NP-Complete problem"

The [language to use here is not straight forward](https://english.stackexchange.com/questions/1016/do-you-use-a-or-an-before-acronyms-initialisms). "is an" reads more clearly.